torchmetrics/metric.py

```diff
@@ -451,6 +451,10 @@ def __pos__(self):
     def __getitem__(self, idx):
         return CompositionalMetric(lambda x: x[idx], self, None)
 
+    @property
+    def is_differentiable(self):
+        raise NotImplementedError
```

Comment:
please add test to this file https://github.com/PyTorchLightning/metrics/blob/master/tests/bases/test_metric.py that the error is raised if a custom metric does not implement this property

Reply:
If i add this,
```python
def test_differentiability():
    with pytest.raises(NotImplementedError):
        metric = DummyMetric()
        metric.is_differentiable
```
Is it seems fine?

Reply:
Yes, I think it is fine.

Reply:
shall we rather have by default None?